Macros/MFIS_read_update_file.sas

```diff
@@ -0,0 +1,321 @@
+/**************************************************************************
+ Program:  MFIS_read_update_file.sas
+ Library:  HUD
+ Project:  NeighborhoodInfo DC
+ Author:   P. Tatian
+ Created:  09/19/15
+ Version:  SAS 9.2
+ Environment:  Local Windows session (desktop)
+ 
+ Description:  Autocall macro to read HUD MFIS update file.
+
+ Modifications:
+**************************************************************************/
+
+/** Macro MFIS_read_update_file - Start Definition **/
+
+%macro MFIS_read_update_file( 
+  filedate=,                      /** File extract date (SAS date value) **/
+  folder=&_dcdata_r_path\HUD,     /** Folder for input raw files **/ 
+  finalize=N,                     /** Upload and register file with metadata (Y/N) **/
+  revisions=%str(New file.)       /** Metadata revision description **/
+  );
+  
+  %let finalize = %upcase( &finalize );
+  
+  %if &finalize = Y and not &_REMOTE_BATCH_SUBMIT %then %do;
+    %warn_mput( macro=MFIS_read_update_file,
+                msg=Not remote batch submit session. Finalize will be set to N. )
+    %let finalize = N;
+  %end;
+ 
+  %local month year filedate_fmt ds_label;
+
+  %let month = %sysfunc( month( &filedate ), z2. );
+  %let year  = %sysfunc( year( &filedate ), z4. );
+  %let filedate_fmt = %sysfunc( putn( &filedate, mmddyyd10. ) );
+  %let ds_label = HUD-insured mortgages, &filedate_fmt update; 
+  
+  data Active;
+  
+    infile "&folder\raw\mfis\rm-a_&filedate_fmt..csv" dsd stopover lrecl=2000 firstobs=2;
+
+    input 
+      HUD_project_number : $40.
+      Premise_id : $16.
+      Property_name : $40.
+      Property_street : $40.
+      Property_city : $40.
+      Property_state : $2.
+      Property_zip : $5.
+      Units : 8.
+      Initial_endorsement_date : mmddyy10.
+      Final_endorsement_date : mmddyy10. 
+      Original_mortgage_amount : comma24.
+      First_payment_date : mmddyy10.
+      Maturity_date : mmddyy10.
+      Term_in_months : 8.
+      Interest_rate : 8.2    
+      Current_principal_and_interest : comma24.2
+      Amortized_principal_balance :  comma24.2
+      Holder_name : $40.
+      Holder_city : $40.
+      Holder_state : $2.
+      Servicer_name : $40.
+      Servicer_city : $40.
+      Servicer_state : $2.
+      SOA_code : $3.
+      _SOA_cat_sub_cat : $80.
+      _TE : $2.
+      _TC : $2.;
+
+    if Property_state in ( 'DC', 'MD', 'VA', 'WV' );
+  
+  run;
+  
+  data Terminated;
```

Comment:
Assumes that both an active and terminated data set are available. Need to change if that's not always the case.